subsys/net/ip/tcp2.c

```diff
@@ -208,6 +210,11 @@ static void tcp_send_queue_flush(struct tcp *conn)
 				struct net_pkt, next))) {
 		tcp_pkt_unref(pkt);
 	}
+
+	if (conn->pending_data) {
+		net_pkt_unref(conn->pending_data);
```

Comment:
Let's use tcp_pkt_unref()?

Reply:
As we did not allocate the pending_data using tcp_pkt_alloc() we cannot use tcp_pkt_unref() here. Not sure if this is a big issue as the conn->pending_data is there for the whole duration of the connection. So it is not allocated/freed constantly.